### DISCUSSION TITLE
Ability to flash the rx through the Ardupilot pass-through

### DISCUSSION THREAD

**Discussion by robustini** [ExpressLRS/ExpressLRS#1569] [Ideas]
Ardupilot is known to allow [Serial Passthrough](https://ardupilot.org/copter/docs/common-serial-passthrough.html) on any serial in flight controls that support it.
This allows multiple things, such as configuring gps, telemetry and rx flash.
It would therefore be useful if the Configurator had the possibility of being able to flash the rx using this system, and not just looking for Betaflight and its DFU.
Not everyone uses ExpressLRS with fliight controls that support Betaflight or INAV, with Ardupilot it works well, I'm testing it and I have no problems, but having to flash the rx using another fc is inconvenient imho.

**Comment by robustini** [comment]
Leaving Ardupilot  "SERIALx_PROTOCOL" parameters at 23 and not at -1 (as theoretically it would be necessary) it finds the bootloader as seen in the log but obviously then it does not communicate to us.
"SERIALx_PROTOCOL" would like to be correctly set to -1, but in that case the Configurator is unable to put the rx in the bootloader in any way.
[ExpressLRSBuildLog_20220513120946846.txt](https://github.com/ExpressLRS/ExpressLRS/files/8687014/ExpressLRSBuildLog_20220513120946846.txt)

**Comment by robustini** [comment]
I tried again and it works leaving the protocol at 23, inexplicable!

**Reply by Jgunde** [reply]
Would you be able to explain how you got this working?
I'm still unable to flash my BetaFPV SuperD RX using Betaflight Passthrough.
I have it connected to my FC on serial 6. Here are the ArduPilot Params:
SERIAL6_BAUD: 460800
SERIAL6_OPTIONS: 0
SERIAL6_PROTOCOL: 23
SERIAL_PASSTIMO: 0
SERIAL_PASS2: Serial6

I then tried to upload the firmware using the Betaflight Passthrough mode.
I tired with the receiver in normal and in bootloader mode, but neither worked.

**Comment by seankovacs** [comment]
Any updates on this?
